csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.shared.cs

```diff
@@ -161,10 +161,33 @@ public static SessionOptions MakeSessionOptionWithTvmProvider(String settings =
         /// <returns>A SessionsOptions() object configured for execution on deviceId</returns>
         public static SessionOptions MakeSessionOptionWithRocmProvider(int deviceId = 0)
         {
+            CheckRocmExecutionProviderDLLs();
             SessionOptions options = new SessionOptions();
-            options.AppendExecutionProvider_ROCM(deviceId);
+            options.AppendExecutionProvider_ROCm(deviceId);
```

Comment:
>options.AppendExecutionProvider_ROCm(deviceId); [](http://example.com/codeflow?start=12&length=47)

Should Append* throw, options would leak. Please, Dispose() on exception

Reply:
Done. Other impls seems also need an update.

Reply:
See https://github.com/microsoft/onnxruntime/pull/15973 for other impls' update.